geode-for-redis/src/distributedTest/java/org/apache/geode/redis/internal/commands/executor/list/LPopDUnitTest.java

```diff
@@ -141,20 +141,22 @@ private void lpushPerformAndVerify(String key, List<String> elementList) {
 
   private void lpopPerformAndVerify(String key, AtomicLong runningCount) {
     assertThat(jedis.llen(key)).isEqualTo(INITIAL_LIST_SIZE);
-    assertThat(jedis.llen(key)).isEqualTo(INITIAL_LIST_SIZE);
 
     int elementCount = INITIAL_LIST_SIZE - 1;
     while (jedis.llen(key) > 0 && runningCount.get() > 0) {
-      String expected = key + "-" + (elementCount - 1) + "-";
-      String element = expected;
+      // String expected = makeElementString(key, elementCount);
+      String element = "";
       try {
         element = jedis.lpop(key);
-        if (!element.equals(expected)) {
-          int expectedCount = elementCount - 1;
-          String[] chunks = element.split("-");
-          elementCount = Integer.parseInt(chunks[4]);
-          assertThat(expectedCount - elementCount).isLessThanOrEqualTo(1);
-        }
+        String[] chunks = element.split("-");
+        int actualCount = Integer.parseInt(chunks[4]);
+        int expectedCount = elementCount;
+        assertThat(actualCount)
+            .as("Actual element count did not match expected element count for element " + element)
+            .satisfiesAnyOf(
+                count -> assertThat(count).isEqualTo(expectedCount),
+                count -> assertThat(count).isEqualTo(expectedCount - 1));
```

Comment:
Similar comment to the RPop dunit - is it necessary to have multiple expectations? Is it not more deterministic?